Error when executing operations using entry fields

To report a bug, fill in the information below. 
For support and feature requests, please use the discussion forum: 
https://discuss.hail.is/

Please include the full Hail version and as much detail as possible.

-----------------------------------------------------------------------------

An error arises when executing operations/actions using field entries. For intance executing  `mt.AD.show()` would give the following: 

### Error No.1: 
```python
---------------------------------------------------------------------------
FatalError                                Traceback (most recent call last)
/usr/local/lib/python3.6/site-packages/IPython/core/formatters.py in __call__(self, obj)
    700                 type_pprinters=self.type_printers,
    701                 deferred_pprinters=self.deferred_printers)
--> 702             printer.pretty(obj)
    703             printer.flush()
    704             return stream.getvalue()

/usr/local/lib/python3.6/site-packages/IPython/lib/pretty.py in pretty(self, obj)
    400                         if cls is not object \
    401                                 and callable(cls.__dict__.get('__repr__')):
--> 402                             return _repr_pprint(obj, self, cycle)
    403 
    404             return _default_pprint(obj, self, cycle)

/usr/local/lib/python3.6/site-packages/IPython/lib/pretty.py in _repr_pprint(obj, p, cycle)
    695     """A pprint that just redirects to the normal repr function."""
    696     # Find newlines and replace them with p.break_()
--> 697     output = repr(obj)
    698     for idx,output_line in enumerate(output.splitlines()):
    699         if idx:

/usr/local/lib/python3.6/site-packages/hail/matrixtable.py in __repr__(self)
   2521 
   2522         def __repr__(self):
-> 2523             return self.__str__()
   2524 
   2525         def _repr_html_(self):

/usr/local/lib/python3.6/site-packages/hail/matrixtable.py in __str__(self)
   2515 
   2516         def __str__(self):
-> 2517             s = self.table_show.__str__()
   2518             if self.displayed_n_cols != self.actual_n_cols:
   2519                 s += f"showing the first { self.displayed_n_cols } of { self.actual_n_cols } columns"

/usr/local/lib/python3.6/site-packages/hail/table.py in __str__(self)
   1241 
   1242         def __str__(self):
-> 1243             return self._ascii_str()
   1244 
   1245         def __repr__(self):

/usr/local/lib/python3.6/site-packages/hail/table.py in _ascii_str(self)
   1268                     return s
   1269 
-> 1270             rows, has_more, dtype = self.data()
   1271             fields = list(dtype)
   1272             trunc_fields = [trunc(f) for f in fields]

/usr/local/lib/python3.6/site-packages/hail/table.py in data(self)
   1251                 row_dtype = t.row.dtype
   1252                 t = t.select(**{k: Table._hl_format(v, self.truncate) for (k, v) in t.row.items()})
-> 1253                 rows, has_more = t._take_n(self.n)
   1254                 self._data = (rows, has_more, row_dtype)
   1255             return self._data

/usr/local/lib/python3.6/site-packages/hail/table.py in _take_n(self, n)
   1372             has_more = False
   1373         else:
-> 1374             rows = self.take(n + 1)
   1375             has_more = len(rows) > n
   1376             rows = rows[:n]

</usr/local/lib/python3.6/site-packages/decorator.py:decorator-gen-1029> in take(self, n, _localize)

/usr/local/lib/python3.6/site-packages/hail/typecheck/check.py in wrapper(__original_func, *args, **kwargs)
    583     def wrapper(__original_func, *args, **kwargs):
    584         args_, kwargs_ = check_all(__original_func, args, kwargs, checkers, is_method=is_method)
--> 585         return __original_func(*args_, **kwargs_)
    586 
    587     return wrapper

/usr/local/lib/python3.6/site-packages/hail/table.py in take(self, n, _localize)
   2011         """
   2012 
-> 2013         return self.head(n).collect(_localize)
   2014 
   2015     @typecheck_method(n=int)

</usr/local/lib/python3.6/site-packages/decorator.py:decorator-gen-1023> in collect(self, _localize)

/usr/local/lib/python3.6/site-packages/hail/typecheck/check.py in wrapper(__original_func, *args, **kwargs)
    583     def wrapper(__original_func, *args, **kwargs):
    584         args_, kwargs_ = check_all(__original_func, args, kwargs, checkers, is_method=is_method)
--> 585         return __original_func(*args_, **kwargs_)
    586 
    587     return wrapper

/usr/local/lib/python3.6/site-packages/hail/table.py in collect(self, _localize)
   1825         e = construct_expr(ir, hl.tarray(self.row.dtype))
   1826         if _localize:
-> 1827             return Env.backend().execute(e._ir)
   1828         else:
   1829             return e

/usr/local/lib/python3.6/site-packages/hail/backend/backend.py in execute(self, ir, timed)
    106 
    107     def execute(self, ir, timed=False):
--> 108         result = json.loads(Env.hc()._jhc.backend().executeJSON(self._to_java_ir(ir)))
    109         value = ir.typ._from_json(result['value'])
    110         timings = result['timings']

/usr/lib/spark/python/lib/py4j-src.zip/py4j/java_gateway.py in __call__(self, *args)
   1255         answer = self.gateway_client.send_command(command)
   1256         return_value = get_return_value(
-> 1257             answer, self.gateway_client, self.target_id, self.name)
   1258 
   1259         for temp_arg in temp_args:

/usr/local/lib/python3.6/site-packages/hail/utils/java.py in deco(*args, **kwargs)
    223             raise FatalError('%s\n\nJava stack trace:\n%s\n'
    224                              'Hail version: %s\n'
--> 225                              'Error summary: %s' % (deepest, full, hail.__version__, deepest)) from None
    226         except pyspark.sql.utils.CapturedException as e:
    227             raise FatalError('%s\n\nJava stack trace:\n%s\n'

FatalError: NoSuchElementException: key not found: GRCh37

```

### Traces No. 1: 

```java 
Java stack trace:
org.apache.spark.SparkException: Job aborted due to stage failure: Task 15 in stage 16.0 failed 4 times, most recent failure: Lost task 15.3 in stage 16.0 (TID 178, ip-172-31-1-20.ec2.internal, executor 4): org.json4s.package$MappingException: unknown error
	at org.json4s.Extraction$.extract(Extraction.scala:43)
	at org.json4s.ExtractableJsonAstNode.extract(ExtractableJsonAstNode.scala:21)
	at is.hail.io.index.IndexReader$.readMetadata(IndexReader.scala:65)
	at is.hail.io.index.IndexReader.<init>(IndexReader.scala:90)
	at is.hail.io.index.IndexReaderBuilder$$anonfun$withDecoders$1.apply(IndexReader.scala:50)
	at is.hail.io.index.IndexReaderBuilder$$anonfun$withDecoders$1.apply(IndexReader.scala:50)
	at is.hail.HailContext$$anon$3$$anonfun$20.apply(HailContext.scala:879)
	at is.hail.HailContext$$anon$3$$anonfun$20.apply(HailContext.scala:877)
	at scala.Option.map(Option.scala:146)
	at is.hail.HailContext$$anon$3.compute(HailContext.scala:877)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:55)
	at org.apache.spark.scheduler.Task.run(Task.scala:121)
	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:402)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:408)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.util.NoSuchElementException: key not found: GRCh37
	at scala.collection.MapLike$class.default(MapLike.scala:228)
	at scala.collection.AbstractMap.default(Map.scala:59)
	at scala.collection.MapLike$class.apply(MapLike.scala:141)
	at scala.collection.AbstractMap.apply(Map.scala:59)
	at is.hail.expr.ir.TypeParserEnvironment.getReferenceGenome(Parser.scala:136)
	at is.hail.expr.ir.IRParser$.type_expr(Parser.scala:363)
	at is.hail.expr.ir.IRParser$.type_field(Parser.scala:324)
	at is.hail.expr.ir.IRParser$$anonfun$13.apply(Parser.scala:406)
	at is.hail.expr.ir.IRParser$$anonfun$13.apply(Parser.scala:406)
	at is.hail.expr.ir.IRParser$.repsepUntil(Parser.scala:289)
	at is.hail.expr.ir.IRParser$.type_expr(Parser.scala:406)
	at is.hail.expr.ir.IRParser$$anonfun$parseType$1.apply(Parser.scala:1443)
	at is.hail.expr.ir.IRParser$$anonfun$parseType$1.apply(Parser.scala:1443)
	at is.hail.expr.ir.IRParser$.parse(Parser.scala:1419)
	at is.hail.expr.ir.IRParser$.parseType(Parser.scala:1443)
	at is.hail.expr.ir.IRParser$.parseType(Parser.scala:1457)
	at is.hail.expr.types.virtual.TypeSerializer$$anonfun$$lessinit$greater$1$$anonfun$apply$1.applyOrElse(Type.scala:19)
	at is.hail.expr.types.virtual.TypeSerializer$$anonfun$$lessinit$greater$1$$anonfun$apply$1.applyOrElse(Type.scala:19)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:402)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:400)
	at scala.PartialFunction$OrElse.apply(PartialFunction.scala:167)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:400)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:400)
	at scala.PartialFunction$OrElse.apply(PartialFunction.scala:167)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:400)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:400)
	at scala.PartialFunction$OrElse.apply(PartialFunction.scala:167)
	at scala.PartialFunction$class.applyOrElse(PartialFunction.scala:123)
	at scala.collection.AbstractMap.applyOrElse(Map.scala:59)
	at scala.PartialFunction$OrElse.apply(PartialFunction.scala:167)
	at org.json4s.Extraction$.customOrElse(Extraction.scala:605)
	at org.json4s.Extraction$.extract(Extraction.scala:392)
	at org.json4s.Extraction$ClassInstanceBuilder.org$json4s$Extraction$ClassInstanceBuilder$$buildCtorArg(Extraction.scala:514)
	at org.json4s.Extraction$ClassInstanceBuilder$$anonfun$15.apply(Extraction.scala:546)
	at org.json4s.Extraction$ClassInstanceBuilder$$anonfun$15.apply(Extraction.scala:546)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
	at org.json4s.Extraction$ClassInstanceBuilder.instantiate(Extraction.scala:546)
	at org.json4s.Extraction$ClassInstanceBuilder.result(Extraction.scala:597)
	at org.json4s.Extraction$$anonfun$extract$6.apply(Extraction.scala:400)
	at org.json4s.Extraction$$anonfun$extract$6.apply(Extraction.scala:392)
	at org.json4s.Extraction$.customOrElse(Extraction.scala:606)
	at org.json4s.Extraction$.extract(Extraction.scala:392)
	at org.json4s.Extraction$.extract(Extraction.scala:39)
	... 38 more

Driver stacktrace:
	at org.apache.spark.scheduler.DAGScheduler.org$apache$spark$scheduler$DAGScheduler$$failJobAndIndependentStages(DAGScheduler.scala:2039)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:2027)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:2026)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2026)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:966)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:966)
	at scala.Option.foreach(Option.scala:257)
	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:966)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2260)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2209)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2198)
	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:777)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2061)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2082)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2101)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2126)
	at org.apache.spark.rdd.RDD$$anonfun$collect$1.apply(RDD.scala:945)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
	at org.apache.spark.rdd.RDD.withScope(RDD.scala:363)
	at org.apache.spark.rdd.RDD.collect(RDD.scala:944)
	at is.hail.sparkextras.ContextRDD.collect(ContextRDD.scala:222)
	at is.hail.rvd.RVD$.getKeyInfo(RVD.scala:1227)
	at is.hail.rvd.RVD.coalesce(RVD.scala:281)
	at is.hail.expr.ir.TableRepartition.execute(TableIR.scala:482)
	at is.hail.expr.ir.TableMapRows.execute(TableIR.scala:929)
	at is.hail.expr.ir.TableRename.execute(TableIR.scala:1630)
	at is.hail.expr.ir.TableKeyBy.execute(TableIR.scala:354)
	at is.hail.expr.ir.TableMapRows.execute(TableIR.scala:929)
	at is.hail.expr.ir.TableMapGlobals.execute(TableIR.scala:1083)
	at is.hail.expr.ir.TableOrderBy.execute(TableIR.scala:1559)
	at is.hail.expr.ir.TableHead.execute(TableIR.scala:458)
	at is.hail.expr.ir.TableMapRows.execute(TableIR.scala:929)
	at is.hail.expr.ir.Interpret$.apply(Interpret.scala:721)
	at is.hail.expr.ir.Interpret$.apply(Interpret.scala:91)
	at is.hail.expr.ir.Interpret$.apply(Interpret.scala:61)
	at is.hail.expr.ir.InterpretNonCompilable$$anonfun$5.apply(InterpretNonCompilable.scala:16)
	at is.hail.expr.ir.InterpretNonCompilable$$anonfun$5.apply(InterpretNonCompilable.scala:16)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.mutable.ArrayOps$ofRef.map(ArrayOps.scala:186)
	at is.hail.expr.ir.InterpretNonCompilable$.apply(InterpretNonCompilable.scala:16)
	at is.hail.expr.ir.CompileAndEvaluate$$anonfun$2.apply(CompileAndEvaluate.scala:37)
	at is.hail.expr.ir.CompileAndEvaluate$$anonfun$2.apply(CompileAndEvaluate.scala:37)
	at is.hail.utils.ExecutionTimer.time(ExecutionTimer.scala:24)
	at is.hail.expr.ir.CompileAndEvaluate$.apply(CompileAndEvaluate.scala:37)
	at is.hail.backend.Backend$$anonfun$execute$1.apply(Backend.scala:56)
	at is.hail.backend.Backend$$anonfun$execute$1.apply(Backend.scala:56)
	at is.hail.expr.ir.ExecuteContext$$anonfun$scoped$1.apply(ExecuteContext.scala:8)
	at is.hail.expr.ir.ExecuteContext$$anonfun$scoped$1.apply(ExecuteContext.scala:7)
	at is.hail.utils.package$.using(package.scala:596)
	at is.hail.annotations.Region$.scoped(Region.scala:18)
	at is.hail.expr.ir.ExecuteContext$.scoped(ExecuteContext.scala:7)
	at is.hail.backend.Backend.execute(Backend.scala:56)
	at is.hail.backend.Backend.executeJSON(Backend.scala:62)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:282)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:238)
	at java.lang.Thread.run(Thread.java:748)

org.json4s.package$MappingException: unknown error
	at org.json4s.Extraction$.extract(Extraction.scala:43)
	at org.json4s.ExtractableJsonAstNode.extract(ExtractableJsonAstNode.scala:21)
	at is.hail.io.index.IndexReader$.readMetadata(IndexReader.scala:65)
	at is.hail.io.index.IndexReader.<init>(IndexReader.scala:90)
	at is.hail.io.index.IndexReaderBuilder$$anonfun$withDecoders$1.apply(IndexReader.scala:50)
	at is.hail.io.index.IndexReaderBuilder$$anonfun$withDecoders$1.apply(IndexReader.scala:50)
	at is.hail.HailContext$$anon$3$$anonfun$20.apply(HailContext.scala:879)
	at is.hail.HailContext$$anon$3$$anonfun$20.apply(HailContext.scala:877)
	at scala.Option.map(Option.scala:146)
	at is.hail.HailContext$$anon$3.compute(HailContext.scala:877)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:55)
	at org.apache.spark.scheduler.Task.run(Task.scala:121)
	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:402)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:408)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

java.util.NoSuchElementException: key not found: GRCh37
	at scala.collection.MapLike$class.default(MapLike.scala:228)
	at scala.collection.AbstractMap.default(Map.scala:59)
	at scala.collection.MapLike$class.apply(MapLike.scala:141)
	at scala.collection.AbstractMap.apply(Map.scala:59)
	at is.hail.expr.ir.TypeParserEnvironment.getReferenceGenome(Parser.scala:136)
	at is.hail.expr.ir.IRParser$.type_expr(Parser.scala:363)
	at is.hail.expr.ir.IRParser$.type_field(Parser.scala:324)
	at is.hail.expr.ir.IRParser$$anonfun$13.apply(Parser.scala:406)
	at is.hail.expr.ir.IRParser$$anonfun$13.apply(Parser.scala:406)
	at is.hail.expr.ir.IRParser$.repsepUntil(Parser.scala:289)
	at is.hail.expr.ir.IRParser$.type_expr(Parser.scala:406)
	at is.hail.expr.ir.IRParser$$anonfun$parseType$1.apply(Parser.scala:1443)
	at is.hail.expr.ir.IRParser$$anonfun$parseType$1.apply(Parser.scala:1443)
	at is.hail.expr.ir.IRParser$.parse(Parser.scala:1419)
	at is.hail.expr.ir.IRParser$.parseType(Parser.scala:1443)
	at is.hail.expr.ir.IRParser$.parseType(Parser.scala:1457)
	at is.hail.expr.types.virtual.TypeSerializer$$anonfun$$lessinit$greater$1$$anonfun$apply$1.applyOrElse(Type.scala:19)
	at is.hail.expr.types.virtual.TypeSerializer$$anonfun$$lessinit$greater$1$$anonfun$apply$1.applyOrElse(Type.scala:19)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:402)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:400)
	at scala.PartialFunction$OrElse.apply(PartialFunction.scala:167)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:400)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:400)
	at scala.PartialFunction$OrElse.apply(PartialFunction.scala:167)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:400)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:400)
	at scala.PartialFunction$OrElse.apply(PartialFunction.scala:167)
	at scala.PartialFunction$class.applyOrElse(PartialFunction.scala:123)
	at scala.collection.AbstractMap.applyOrElse(Map.scala:59)
	at scala.PartialFunction$OrElse.apply(PartialFunction.scala:167)
	at org.json4s.Extraction$.customOrElse(Extraction.scala:605)
	at org.json4s.Extraction$.extract(Extraction.scala:392)
	at org.json4s.Extraction$ClassInstanceBuilder.org$json4s$Extraction$ClassInstanceBuilder$$buildCtorArg(Extraction.scala:514)
	at org.json4s.Extraction$ClassInstanceBuilder$$anonfun$15.apply(Extraction.scala:546)
	at org.json4s.Extraction$ClassInstanceBuilder$$anonfun$15.apply(Extraction.scala:546)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
	at org.json4s.Extraction$ClassInstanceBuilder.instantiate(Extraction.scala:546)
	at org.json4s.Extraction$ClassInstanceBuilder.result(Extraction.scala:597)
	at org.json4s.Extraction$$anonfun$extract$6.apply(Extraction.scala:400)
	at org.json4s.Extraction$$anonfun$extract$6.apply(Extraction.scala:392)
	at org.json4s.Extraction$.customOrElse(Extraction.scala:606)
	at org.json4s.Extraction$.extract(Extraction.scala:392)
	at org.json4s.Extraction$.extract(Extraction.scala:39)
	at org.json4s.ExtractableJsonAstNode.extract(ExtractableJsonAstNode.scala:21)
	at is.hail.io.index.IndexReader$.readMetadata(IndexReader.scala:65)
	at is.hail.io.index.IndexReader.<init>(IndexReader.scala:90)
	at is.hail.io.index.IndexReaderBuilder$$anonfun$withDecoders$1.apply(IndexReader.scala:50)
	at is.hail.io.index.IndexReaderBuilder$$anonfun$withDecoders$1.apply(IndexReader.scala:50)
	at is.hail.HailContext$$anon$3$$anonfun$20.apply(HailContext.scala:879)
	at is.hail.HailContext$$anon$3$$anonfun$20.apply(HailContext.scala:877)
	at scala.Option.map(Option.scala:146)
	at is.hail.HailContext$$anon$3.compute(HailContext.scala:877)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:55)
	at org.apache.spark.scheduler.Task.run(Task.scala:121)
	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:402)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:408)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)


Hail version: 0.2.21-1d317a44e5fd
Error summary: NoSuchElementException: key not found: GRCh37
```

### Error No. 2
```python
---------------------------------------------------------------------------
FatalError                                Traceback (most recent call last)
/usr/local/lib/python3.6/site-packages/IPython/core/formatters.py in __call__(self, obj)
    343             method = get_real_method(obj, self.print_method)
    344             if method is not None:
--> 345                 return method()
    346             return None
    347         else:

/usr/local/lib/python3.6/site-packages/hail/matrixtable.py in _repr_html_(self)
   2524 
   2525         def _repr_html_(self):
-> 2526             s = self.table_show._repr_html_()
   2527             if self.displayed_n_cols != self.actual_n_cols:
   2528                 s += '<p style="background: #fdd; padding: 0.4em;">'

/usr/local/lib/python3.6/site-packages/hail/table.py in _repr_html_(self)
   1256 
   1257         def _repr_html_(self):
-> 1258             return self._html_str()
   1259 
   1260         def _ascii_str(self):

/usr/local/lib/python3.6/site-packages/hail/table.py in _html_str(self)
   1342             types = self.types
   1343 
-> 1344             rows, has_more, dtype = self.data()
   1345             fields = list(dtype)
   1346 

/usr/local/lib/python3.6/site-packages/hail/table.py in data(self)
   1251                 row_dtype = t.row.dtype
   1252                 t = t.select(**{k: Table._hl_format(v, self.truncate) for (k, v) in t.row.items()})
-> 1253                 rows, has_more = t._take_n(self.n)
   1254                 self._data = (rows, has_more, row_dtype)
   1255             return self._data

/usr/local/lib/python3.6/site-packages/hail/table.py in _take_n(self, n)
   1372             has_more = False
   1373         else:
-> 1374             rows = self.take(n + 1)
   1375             has_more = len(rows) > n
   1376             rows = rows[:n]

</usr/local/lib/python3.6/site-packages/decorator.py:decorator-gen-1029> in take(self, n, _localize)

/usr/local/lib/python3.6/site-packages/hail/typecheck/check.py in wrapper(__original_func, *args, **kwargs)
    583     def wrapper(__original_func, *args, **kwargs):
    584         args_, kwargs_ = check_all(__original_func, args, kwargs, checkers, is_method=is_method)
--> 585         return __original_func(*args_, **kwargs_)
    586 
    587     return wrapper

/usr/local/lib/python3.6/site-packages/hail/table.py in take(self, n, _localize)
   2011         """
   2012 
-> 2013         return self.head(n).collect(_localize)
   2014 
   2015     @typecheck_method(n=int)

</usr/local/lib/python3.6/site-packages/decorator.py:decorator-gen-1023> in collect(self, _localize)

/usr/local/lib/python3.6/site-packages/hail/typecheck/check.py in wrapper(__original_func, *args, **kwargs)
    583     def wrapper(__original_func, *args, **kwargs):
    584         args_, kwargs_ = check_all(__original_func, args, kwargs, checkers, is_method=is_method)
--> 585         return __original_func(*args_, **kwargs_)
    586 
    587     return wrapper

/usr/local/lib/python3.6/site-packages/hail/table.py in collect(self, _localize)
   1825         e = construct_expr(ir, hl.tarray(self.row.dtype))
   1826         if _localize:
-> 1827             return Env.backend().execute(e._ir)
   1828         else:
   1829             return e

/usr/local/lib/python3.6/site-packages/hail/backend/backend.py in execute(self, ir, timed)
    106 
    107     def execute(self, ir, timed=False):
--> 108         result = json.loads(Env.hc()._jhc.backend().executeJSON(self._to_java_ir(ir)))
    109         value = ir.typ._from_json(result['value'])
    110         timings = result['timings']

/usr/lib/spark/python/lib/py4j-src.zip/py4j/java_gateway.py in __call__(self, *args)
   1255         answer = self.gateway_client.send_command(command)
   1256         return_value = get_return_value(
-> 1257             answer, self.gateway_client, self.target_id, self.name)
   1258 
   1259         for temp_arg in temp_args:

/usr/local/lib/python3.6/site-packages/hail/utils/java.py in deco(*args, **kwargs)
    223             raise FatalError('%s\n\nJava stack trace:\n%s\n'
    224                              'Hail version: %s\n'
--> 225                              'Error summary: %s' % (deepest, full, hail.__version__, deepest)) from None
    226         except pyspark.sql.utils.CapturedException as e:
    227             raise FatalError('%s\n\nJava stack trace:\n%s\n'

FatalError: NoSuchElementException: key not found: GRCh37
```

### Traces No.2:
```java
Java stack trace:
org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 19.0 failed 4 times, most recent failure: Lost task 0.3 in stage 19.0 (TID 220, ip-172-31-2-255.ec2.internal, executor 2): org.json4s.package$MappingException: unknown error
	at org.json4s.Extraction$.extract(Extraction.scala:43)
	at org.json4s.ExtractableJsonAstNode.extract(ExtractableJsonAstNode.scala:21)
	at is.hail.io.index.IndexReader$.readMetadata(IndexReader.scala:65)
	at is.hail.io.index.IndexReader.<init>(IndexReader.scala:90)
	at is.hail.io.index.IndexReaderBuilder$$anonfun$withDecoders$1.apply(IndexReader.scala:50)
	at is.hail.io.index.IndexReaderBuilder$$anonfun$withDecoders$1.apply(IndexReader.scala:50)
	at is.hail.HailContext$$anon$3$$anonfun$20.apply(HailContext.scala:879)
	at is.hail.HailContext$$anon$3$$anonfun$20.apply(HailContext.scala:877)
	at scala.Option.map(Option.scala:146)
	at is.hail.HailContext$$anon$3.compute(HailContext.scala:877)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:55)
	at org.apache.spark.scheduler.Task.run(Task.scala:121)
	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:402)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:408)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.util.NoSuchElementException: key not found: GRCh37
	at scala.collection.MapLike$class.default(MapLike.scala:228)
	at scala.collection.AbstractMap.default(Map.scala:59)
	at scala.collection.MapLike$class.apply(MapLike.scala:141)
	at scala.collection.AbstractMap.apply(Map.scala:59)
	at is.hail.expr.ir.TypeParserEnvironment.getReferenceGenome(Parser.scala:136)
	at is.hail.expr.ir.IRParser$.type_expr(Parser.scala:363)
	at is.hail.expr.ir.IRParser$.type_field(Parser.scala:324)
	at is.hail.expr.ir.IRParser$$anonfun$13.apply(Parser.scala:406)
	at is.hail.expr.ir.IRParser$$anonfun$13.apply(Parser.scala:406)
	at is.hail.expr.ir.IRParser$.repsepUntil(Parser.scala:289)
	at is.hail.expr.ir.IRParser$.type_expr(Parser.scala:406)
	at is.hail.expr.ir.IRParser$$anonfun$parseType$1.apply(Parser.scala:1443)
	at is.hail.expr.ir.IRParser$$anonfun$parseType$1.apply(Parser.scala:1443)
	at is.hail.expr.ir.IRParser$.parse(Parser.scala:1419)
	at is.hail.expr.ir.IRParser$.parseType(Parser.scala:1443)
	at is.hail.expr.ir.IRParser$.parseType(Parser.scala:1457)
	at is.hail.expr.types.virtual.TypeSerializer$$anonfun$$lessinit$greater$1$$anonfun$apply$1.applyOrElse(Type.scala:19)
	at is.hail.expr.types.virtual.TypeSerializer$$anonfun$$lessinit$greater$1$$anonfun$apply$1.applyOrElse(Type.scala:19)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:402)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:400)
	at scala.PartialFunction$OrElse.apply(PartialFunction.scala:167)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:400)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:400)
	at scala.PartialFunction$OrElse.apply(PartialFunction.scala:167)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:400)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:400)
	at scala.PartialFunction$OrElse.apply(PartialFunction.scala:167)
	at scala.PartialFunction$class.applyOrElse(PartialFunction.scala:123)
	at scala.collection.AbstractMap.applyOrElse(Map.scala:59)
	at scala.PartialFunction$OrElse.apply(PartialFunction.scala:167)
	at org.json4s.Extraction$.customOrElse(Extraction.scala:605)
	at org.json4s.Extraction$.extract(Extraction.scala:392)
	at org.json4s.Extraction$ClassInstanceBuilder.org$json4s$Extraction$ClassInstanceBuilder$$buildCtorArg(Extraction.scala:514)
	at org.json4s.Extraction$ClassInstanceBuilder$$anonfun$15.apply(Extraction.scala:546)
	at org.json4s.Extraction$ClassInstanceBuilder$$anonfun$15.apply(Extraction.scala:546)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
	at org.json4s.Extraction$ClassInstanceBuilder.instantiate(Extraction.scala:546)
	at org.json4s.Extraction$ClassInstanceBuilder.result(Extraction.scala:597)
	at org.json4s.Extraction$$anonfun$extract$6.apply(Extraction.scala:400)
	at org.json4s.Extraction$$anonfun$extract$6.apply(Extraction.scala:392)
	at org.json4s.Extraction$.customOrElse(Extraction.scala:606)
	at org.json4s.Extraction$.extract(Extraction.scala:392)
	at org.json4s.Extraction$.extract(Extraction.scala:39)
	... 38 more

Driver stacktrace:
	at org.apache.spark.scheduler.DAGScheduler.org$apache$spark$scheduler$DAGScheduler$$failJobAndIndependentStages(DAGScheduler.scala:2039)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:2027)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:2026)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2026)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:966)
	at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:966)
	at scala.Option.foreach(Option.scala:257)
	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:966)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2260)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2209)
	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2198)
	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:777)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2061)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2082)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2101)
	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2126)
	at org.apache.spark.rdd.RDD$$anonfun$collect$1.apply(RDD.scala:945)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
	at org.apache.spark.rdd.RDD.withScope(RDD.scala:363)
	at org.apache.spark.rdd.RDD.collect(RDD.scala:944)
	at is.hail.sparkextras.ContextRDD.collect(ContextRDD.scala:222)
	at is.hail.rvd.RVD$.getKeyInfo(RVD.scala:1227)
	at is.hail.rvd.RVD.coalesce(RVD.scala:281)
	at is.hail.expr.ir.TableRepartition.execute(TableIR.scala:482)
	at is.hail.expr.ir.TableMapRows.execute(TableIR.scala:929)
	at is.hail.expr.ir.TableRename.execute(TableIR.scala:1630)
	at is.hail.expr.ir.TableKeyBy.execute(TableIR.scala:354)
	at is.hail.expr.ir.TableMapRows.execute(TableIR.scala:929)
	at is.hail.expr.ir.TableMapGlobals.execute(TableIR.scala:1083)
	at is.hail.expr.ir.TableOrderBy.execute(TableIR.scala:1559)
	at is.hail.expr.ir.TableHead.execute(TableIR.scala:458)
	at is.hail.expr.ir.TableMapRows.execute(TableIR.scala:929)
	at is.hail.expr.ir.Interpret$.apply(Interpret.scala:721)
	at is.hail.expr.ir.Interpret$.apply(Interpret.scala:91)
	at is.hail.expr.ir.Interpret$.apply(Interpret.scala:61)
	at is.hail.expr.ir.InterpretNonCompilable$$anonfun$5.apply(InterpretNonCompilable.scala:16)
	at is.hail.expr.ir.InterpretNonCompilable$$anonfun$5.apply(InterpretNonCompilable.scala:16)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
	at scala.collection.mutable.ArrayOps$ofRef.foreach(ArrayOps.scala:186)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.mutable.ArrayOps$ofRef.map(ArrayOps.scala:186)
	at is.hail.expr.ir.InterpretNonCompilable$.apply(InterpretNonCompilable.scala:16)
	at is.hail.expr.ir.CompileAndEvaluate$$anonfun$2.apply(CompileAndEvaluate.scala:37)
	at is.hail.expr.ir.CompileAndEvaluate$$anonfun$2.apply(CompileAndEvaluate.scala:37)
	at is.hail.utils.ExecutionTimer.time(ExecutionTimer.scala:24)
	at is.hail.expr.ir.CompileAndEvaluate$.apply(CompileAndEvaluate.scala:37)
	at is.hail.backend.Backend$$anonfun$execute$1.apply(Backend.scala:56)
	at is.hail.backend.Backend$$anonfun$execute$1.apply(Backend.scala:56)
	at is.hail.expr.ir.ExecuteContext$$anonfun$scoped$1.apply(ExecuteContext.scala:8)
	at is.hail.expr.ir.ExecuteContext$$anonfun$scoped$1.apply(ExecuteContext.scala:7)
	at is.hail.utils.package$.using(package.scala:596)
	at is.hail.annotations.Region$.scoped(Region.scala:18)
	at is.hail.expr.ir.ExecuteContext$.scoped(ExecuteContext.scala:7)
	at is.hail.backend.Backend.execute(Backend.scala:56)
	at is.hail.backend.Backend.executeJSON(Backend.scala:62)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:357)
	at py4j.Gateway.invoke(Gateway.java:282)
	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
	at py4j.commands.CallCommand.execute(CallCommand.java:79)
	at py4j.GatewayConnection.run(GatewayConnection.java:238)
	at java.lang.Thread.run(Thread.java:748)

org.json4s.package$MappingException: unknown error
	at org.json4s.Extraction$.extract(Extraction.scala:43)
	at org.json4s.ExtractableJsonAstNode.extract(ExtractableJsonAstNode.scala:21)
	at is.hail.io.index.IndexReader$.readMetadata(IndexReader.scala:65)
	at is.hail.io.index.IndexReader.<init>(IndexReader.scala:90)
	at is.hail.io.index.IndexReaderBuilder$$anonfun$withDecoders$1.apply(IndexReader.scala:50)
	at is.hail.io.index.IndexReaderBuilder$$anonfun$withDecoders$1.apply(IndexReader.scala:50)
	at is.hail.HailContext$$anon$3$$anonfun$20.apply(HailContext.scala:879)
	at is.hail.HailContext$$anon$3$$anonfun$20.apply(HailContext.scala:877)
	at scala.Option.map(Option.scala:146)
	at is.hail.HailContext$$anon$3.compute(HailContext.scala:877)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:55)
	at org.apache.spark.scheduler.Task.run(Task.scala:121)
	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:402)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:408)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

java.util.NoSuchElementException: key not found: GRCh37
	at scala.collection.MapLike$class.default(MapLike.scala:228)
	at scala.collection.AbstractMap.default(Map.scala:59)
	at scala.collection.MapLike$class.apply(MapLike.scala:141)
	at scala.collection.AbstractMap.apply(Map.scala:59)
	at is.hail.expr.ir.TypeParserEnvironment.getReferenceGenome(Parser.scala:136)
	at is.hail.expr.ir.IRParser$.type_expr(Parser.scala:363)
	at is.hail.expr.ir.IRParser$.type_field(Parser.scala:324)
	at is.hail.expr.ir.IRParser$$anonfun$13.apply(Parser.scala:406)
	at is.hail.expr.ir.IRParser$$anonfun$13.apply(Parser.scala:406)
	at is.hail.expr.ir.IRParser$.repsepUntil(Parser.scala:289)
	at is.hail.expr.ir.IRParser$.type_expr(Parser.scala:406)
	at is.hail.expr.ir.IRParser$$anonfun$parseType$1.apply(Parser.scala:1443)
	at is.hail.expr.ir.IRParser$$anonfun$parseType$1.apply(Parser.scala:1443)
	at is.hail.expr.ir.IRParser$.parse(Parser.scala:1419)
	at is.hail.expr.ir.IRParser$.parseType(Parser.scala:1443)
	at is.hail.expr.ir.IRParser$.parseType(Parser.scala:1457)
	at is.hail.expr.types.virtual.TypeSerializer$$anonfun$$lessinit$greater$1$$anonfun$apply$1.applyOrElse(Type.scala:19)
	at is.hail.expr.types.virtual.TypeSerializer$$anonfun$$lessinit$greater$1$$anonfun$apply$1.applyOrElse(Type.scala:19)
	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:402)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:400)
	at scala.PartialFunction$OrElse.apply(PartialFunction.scala:167)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:400)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:400)
	at scala.PartialFunction$OrElse.apply(PartialFunction.scala:167)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:400)
	at org.json4s.CustomSerializer$$anonfun$deserialize$1.applyOrElse(Formats.scala:400)
	at scala.PartialFunction$OrElse.apply(PartialFunction.scala:167)
	at scala.PartialFunction$class.applyOrElse(PartialFunction.scala:123)
	at scala.collection.AbstractMap.applyOrElse(Map.scala:59)
	at scala.PartialFunction$OrElse.apply(PartialFunction.scala:167)
	at org.json4s.Extraction$.customOrElse(Extraction.scala:605)
	at org.json4s.Extraction$.extract(Extraction.scala:392)
	at org.json4s.Extraction$ClassInstanceBuilder.org$json4s$Extraction$ClassInstanceBuilder$$buildCtorArg(Extraction.scala:514)
	at org.json4s.Extraction$ClassInstanceBuilder$$anonfun$15.apply(Extraction.scala:546)
	at org.json4s.Extraction$ClassInstanceBuilder$$anonfun$15.apply(Extraction.scala:546)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
	at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:48)
	at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
	at scala.collection.AbstractTraversable.map(Traversable.scala:104)
	at org.json4s.Extraction$ClassInstanceBuilder.instantiate(Extraction.scala:546)
	at org.json4s.Extraction$ClassInstanceBuilder.result(Extraction.scala:597)
	at org.json4s.Extraction$$anonfun$extract$6.apply(Extraction.scala:400)
	at org.json4s.Extraction$$anonfun$extract$6.apply(Extraction.scala:392)
	at org.json4s.Extraction$.customOrElse(Extraction.scala:606)
	at org.json4s.Extraction$.extract(Extraction.scala:392)
	at org.json4s.Extraction$.extract(Extraction.scala:39)
	at org.json4s.ExtractableJsonAstNode.extract(ExtractableJsonAstNode.scala:21)
	at is.hail.io.index.IndexReader$.readMetadata(IndexReader.scala:65)
	at is.hail.io.index.IndexReader.<init>(IndexReader.scala:90)
	at is.hail.io.index.IndexReaderBuilder$$anonfun$withDecoders$1.apply(IndexReader.scala:50)
	at is.hail.io.index.IndexReaderBuilder$$anonfun$withDecoders$1.apply(IndexReader.scala:50)
	at is.hail.HailContext$$anon$3$$anonfun$20.apply(HailContext.scala:879)
	at is.hail.HailContext$$anon$3$$anonfun$20.apply(HailContext.scala:877)
	at scala.Option.map(Option.scala:146)
	at is.hail.HailContext$$anon$3.compute(HailContext.scala:877)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:324)
	at org.apache.spark.rdd.RDD.iterator(RDD.scala:288)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:55)
	at org.apache.spark.scheduler.Task.run(Task.scala:121)
	at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:402)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:408)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

Hail version: 0.2.21-1d317a44e5fd
Error summary: NoSuchElementException: key not found: GRCh37
```

Thanks!

--- a/hail/src/main/scala/is/hail/io/index/IndexReader.scala
+++ b/hail/src/main/scala/is/hail/io/index/IndexReader.scala
@@ -5,7 +5,7 @@ import java.util
 import java.util.Map.Entry
 
 import is.hail.annotations._
-import is.hail.expr.types.virtual.Type
+import is.hail.expr.types.virtual.{Type, TypeSerializer}
 import is.hail.expr.ir.IRParser
 import is.hail.expr.types.physical.{PStruct, PType}
 import is.hail.io._
@@ -15,8 +15,8 @@ import is.hail.io.fs.FS
 import is.hail.rvd.AbstractRVDSpec
 import org.apache.hadoop.fs.FSDataInputStream
 import org.apache.spark.sql.Row
-import org.json4s.Formats
-import org.json4s.jackson.JsonMethods
+import org.json4s.{Formats, NoTypeHints}
+import org.json4s.jackson.{JsonMethods, Serialization}
 
 object IndexReaderBuilder {
   def apply(fs: FS, path: String): (FS, String, Int) => IndexReader = {
@@ -59,15 +59,25 @@ object IndexReaderBuilder {
 }
 
 object IndexReader {
-  def readMetadata(fs: FS, path: String): IndexMetadata = {
-    val jv = fs.readFile(path + "/metadata.json.gz") { in => JsonMethods.parse(in) }
-    implicit val formats: Formats = AbstractRVDSpec.formats
-    jv.extract[IndexMetadata]
+  def readUntyped(fs: FS, path: String): IndexMetadataUntypedJSON = {
+    val jv = fs.readFile(path + "/metadata.json.gz") { in =>
+      JsonMethods.parse(in)
+        .removeField{ case (f, _) => f == "keyType" || f == "annotationType" }
+    }
+    implicit val formats: Formats = defaultJSONFormats
+    jv.extract[IndexMetadataUntypedJSON]
+  }
+
+  def readMetadata(fs: FS, path: String, keyType: Type, annotationType: Type): IndexMetadata = {
+    val untyped = IndexReader.readUntyped(fs, path)
+    untyped.toMetadata(keyType, annotationType)
   }
 
   def readTypes(fs: FS, path: String): (Type, Type) = {
-    val metadata = IndexReader.readMetadata(fs, path)
-    (metadata.keyType, metadata.annotationType)
+    val jv = fs.readFile(path + "/metadata.json.gz") { in => JsonMethods.parse(in) }
+    implicit val formats: Formats = defaultJSONFormats + new TypeSerializer
+    val metadata = jv.extract[IndexMetadata]
+    metadata.keyType -> metadata.annotationType
   }
 
   def apply(fs: FS, path: String, cacheCapacity: Int = 8): IndexReader = {
@@ -87,7 +97,7 @@ class IndexReader(fs: FS,
   val leafPType: PStruct,
   val internalPType: PStruct
 ) extends AutoCloseable {
-  private[io] val metadata = IndexReader.readMetadata(fs, path)
+  private[io] val metadata = IndexReader.readMetadata(fs, path, keyType, annotationType)
   val branchingFactor = metadata.branchingFactor
   val height = metadata.height
   val nKeys = metadata.nKeys

--- a/hail/src/main/scala/is/hail/io/index/IndexWriter.scala
+++ b/hail/src/main/scala/is/hail/io/index/IndexWriter.scala
@@ -37,6 +37,21 @@ trait AbstractIndexMetadata {
   def intSpec: CodecSpec2
 }
 
+case class IndexMetadataUntypedJSON(
+  fileVersion: Int,
+  branchingFactor: Int,
+  height: Int,
+  nKeys: Long,
+  indexPath: String,
+  rootOffset: Long,
+  attributes: Map[String, Any]
+) {
+  def toMetadata(keyType: Type, annotationType: Type): IndexMetadata = IndexMetadata(
+    fileVersion, branchingFactor,
+    height, keyType, annotationType,
+    nKeys, indexPath, rootOffset, attributes)
+}
+
 case class IndexMetadata(
   fileVersion: Int,
   branchingFactor: Int,

